kong/db/schema/entities/routes.lua

```diff
@@ -2,18 +2,30 @@ local typedefs = require("kong.db.schema.typedefs")
 local router = require("resty.router.router")
 local deprecation = require("kong.deprecation")
 
-local CACHED_SCHEMA = require("kong.router.atc").schema
-local get_expression = require("kong.router.compat").get_expression
+local validate_entity_by_expression
+local has_paths
+do
+  local isempty        = require("table.isempty")
+  local CACHED_SCHEMA  = require("kong.router.atc").schema
+  local get_expression = require("kong.router.compat").get_expression
 
-local function validate_expression(id, exp)
-  local r = router.new(CACHED_SCHEMA)
+  local type = type
 
-  local res, err = r:add_matcher(0, id, exp)
-  if not res then
-    return nil, "Router Expression failed validation: " .. err
+  validate_entity_by_expression = function(entity)
```

Comment:
```suggestion
  validate_traditional_compatiable_route = function(entity)
```

Might be better illustration of the purpose of this function.

Reply:
rename to `validate_route`